bugsnag-android-core/src/main/java/com/bugsnag/android/InternalReportDelegate.java

```diff
@@ -0,0 +1,134 @@
+        err.addMetadata(INTERNAL_DIAGNOSTICS_TAB, "fileLength", errorFile.length());
+        recordStorageCacheBehavior(err);
+        reportInternalBugsnagError(err);
+    }
+
+    void recordStorageCacheBehavior(Event event) {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            File cacheDir = appContext.getCacheDir();
+            File errDir = new File(cacheDir, "bugsnag-errors");
+
+            try {
+                boolean tombstone = storageManager.isCacheBehaviorTombstone(errDir);
+                boolean group = storageManager.isCacheBehaviorGroup(errDir);
+                event.addMetadata(INTERNAL_DIAGNOSTICS_TAB, "cacheTombstone", tombstone);
+                event.addMetadata(INTERNAL_DIAGNOSTICS_TAB, "cacheGroup", group);
+            } catch (IOException exc) {
+                logger.w("Failed to record cache behaviour, skipping diagnostics", exc);
+            }
+        }
+    }
+
+    /**
+     * Reports an event that occurred within the notifier to bugsnag. A lean event report will be
+     * generated and sent asynchronously with no callbacks, retry attempts, or writing to disk.
+     * This is intended for internal use only, and reports will not be visible to end-users.
+     */
+    void reportInternalBugsnagError(@NonNull Event event) {
+        Map<String, Object> app = appData.getAppDataSummary();
+        app.put("duration", AppData.getDurationMs());
+        app.put("durationInForeground", appData.calculateDurationInForeground());
+        app.put("inForeground", sessionTracker.isInForeground());
+        event.setApp(app);
+
+        Map<String, Object> device = deviceData.getDeviceDataSummary();
+        device.put("freeDisk", deviceData.calculateFreeDisk());
+        event.setDevice(device);
+
+        Notifier notifier = Notifier.INSTANCE;
+        event.addMetadata(INTERNAL_DIAGNOSTICS_TAB, "notifierName", notifier.getName());
+        event.addMetadata(INTERNAL_DIAGNOSTICS_TAB, "notifierVersion", notifier.getVersion());
+        event.addMetadata(INTERNAL_DIAGNOSTICS_TAB, "apiKey", immutableConfig.getApiKey());
+
+        Object packageName = appData.getAppData().get("packageName");
+        event.addMetadata(INTERNAL_DIAGNOSTICS_TAB, "packageName", packageName);
+
+        final Report report = new Report(null, event);
+        try {
+            Async.run(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        Delivery delivery = immutableConfig.getDelivery();
+                        DeliveryParams params = immutableConfig.errorApiDeliveryParams();
+
+                        // can only modify headers if DefaultDelivery is in use
+                        if (delivery instanceof DefaultDelivery) {
+                            Map<String, String> headers = params.getHeaders();
+                            headers.put("Bugsnag-Internal-Error", "true");
```

Comment:
Can you extract this to a header in this class for consistency?

Reply:
Extracted to a header in `ImmutableConfig`.